pkg/auth/authz_handler.go

```diff
@@ -162,6 +270,11 @@ func (a *AuthzHandler) allowedToUseAssistedInstaller(username string) (bool, err
 		context.Background(), username, ocm.AMSActionCreate, ocm.BareMetalClusterResource)
 }
 
+func (a *AuthzHandler) hasClusterEditRole(payload *ocm.AuthPayload, subscriptionID string) (bool, error) {
+	return a.client.Authorization.SubscriptionAccess(
+		context.Background(), payload.Username, payload.Organization, ocm.AMSActionUpdate, subscriptionID)
```

Comment:
So after checking in AMS, seems the AccessReview API accepts an 'AMS id' for the organization (i.e. as for subscription). However, for each user we get in the payload the external_id of the organization, which is different. Any way, the organization ID is optional in this API as it's already available AMS database. Both accounts and subscriptions tables as org_id column, so it's redundant for checking perms. In short, we simply don't need to pass payload.Organization :)

Reply:
fixed